Xamarin.Forms.Platform.Android/Extensions/BrushExtensions.cs

```diff
@@ -163,13 +170,45 @@ public static void UpdateBackground(this GradientDrawable gradientDrawable, Brus
 			}
 		}
 
+		public static void UpdateBackground(this GradientStrokeDrawable gradientStrokeDrawable, Brush brush)
+		{
+			if (gradientStrokeDrawable == null || brush == null || brush.IsEmpty)
+				return;
+
+			gradientStrokeDrawable.SetStroke(0, Color.Default.ToAndroid());
+
+			if (brush is SolidColorBrush solidColorBrush)
+			{
+				var color = solidColorBrush.Color.IsDefault ? Color.Default.ToAndroid() : solidColorBrush.Color.ToAndroid();
+				gradientStrokeDrawable.SetColor(color);
+			}
+			else
+				gradientStrokeDrawable.SetGradient(brush);
+		}
+
 		public static bool UseGradients(this GradientDrawable gradientDrawable)
 		{
 			if (!Forms.IsNougatOrNewer)
 				return false;
 
-			var colors = gradientDrawable.GetColors();
-			return colors != null && colors.Length > 1;
+			if (ApiLevel >= 24)
```

Comment:
Why use a new variable for ApiLevel instead of checking Forms.IsNougatOrNewer?

Reply:
Is not necessary. Changed.